unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Scene/Main/Main.cs

```diff
@@ -107,14 +108,36 @@ protected virtual void SetupServices()
 
         protected virtual void Start()
         {
-            // this event should be the last one to be executed after initialization
-            // it is used by the kernel to signal "EngineReady" or something like that
-            // to prevent race conditions like "SceneController is not an object",
-            // aka sending events before unity is ready
-            WebInterface.SendSystemInfoReport();
-
-            // We trigger the Decentraland logic once everything is initialized.
-            WebInterface.StartDecentraland();
+            var rpc = Environment.i.serviceLocator.Get<IRPC>();
+
+            async void InitializeDecentraland()
+            {
+                await UniTask.SwitchToMainThread();
+
+                // this event should be the last one to be executed after initialization
+                // it is used by the kernel to signal "EngineReady" or something like that
+                // to prevent race conditions like "SceneController is not an object",
+                // aka sending events before unity is ready
+                ClientAnalyticsKernelService analytics = rpc.Analytics();
+
+                await analytics.SystemInfoReport(new SystemInfoReportRequest()
+                {
+                    GraphicsDeviceName = SystemInfo.graphicsDeviceName,
+                    GraphicsDeviceVersion = SystemInfo.graphicsDeviceVersion,
+                    GraphicsMemorySize = (uint)SystemInfo.graphicsMemorySize,
+                    ProcessorType = SystemInfo.processorType,
+                    ProcessorCount = (uint)SystemInfo.processorCount,
+                    SystemMemorySize = (uint)SystemInfo.systemMemorySize,
+
+                    // TODO: remove useBinaryTransform after ECS7 is fully in prod
+                    UseBinaryTransform = true,
+                });
+
+                // We trigger the Decentraland logic once everything is initialized.
+                WebInterface.StartDecentraland();
+            }
+
+            rpc.EnsureRpc().GetAwaiter().OnCompleted(InitializeDecentraland);
```

Comment:
Since this is async already, why do we need the callback? we can await `EnsureRPC` and then do the rest. 

It should also be wrapped in a method  (or a class) somewhere else.